### ISSUE TITLE
Proposal : Into to text classification and NLP

### ISSUE THREAD

**Issue by ishaan007** [pydatadelhi/talks#51]
Every text classification problem in NLP is broadly categorized as a document or a token level classification .This talk will be devoted to document classification

I will walk through a classification experiment I coded and will talk about the process and ins and outs of it.

Slides for the talk : http://bit.ly/2vQC1sc
This is a blog I wrote on all this
https://medium.com/towards-data-science/document-feature-extraction-and-classification-53f0e813d2d3#.lzi3u4t29

Code for the talk
https://github.com/ishaan007/vector_space_modelling

**Comment by manojpandey** [comment]
Is this the one you did at CIC ?

**Comment by ishaan007** [comment]
Yes.

**Comment by manojpandey** [comment]
link to slides ? (:

**Comment by ishaan007** [comment]
Comment updated with the link to the slides.

**Comment by manasRK** [comment]
Could you expand more on what features you used or feature engineering architecture you are proposing? ALso, the talk proposal seems a bit similar to Bag of Words meets Bag of Popcorn on Kaggle. Anything specific on interesting features/architectures would be nice.

Interestingly, your TFIDF has way lower accuracy than embeddings, would be eager to hear why it may have happened.

**Comment by ishaan007** [comment]
@manasRK 

1. TF-IDF has a higher accuracy than embeddings because :
The amount of training data used to train the doc2vec model is way too less for the neural net to learn sensible embeddings. So TF-IDF outperformed it.
2. I will try to explain the custom feature engineering architecture I mentioned in the slides with the below python stub
Say you have to classify tokens in an article(maybe POS tagging) 
So the syntactic features might be 
{
"len_token" :len(sentence.split()[index]) # length of the token 
"is_preposition" :filter(lambda x:x in sentence.split()[index],["the","is","of"]) # whether it is a preposition
}
and contextual features might be
{
"prev_tag" # tag of previous token for context
"next_tag" # tag of next token for context
}
And all these features might be encoded using OHE or feature hashing or any encoding scheme.So both type of features help identify individual tokens .

**Comment by ishaan007** [comment]
Also as far as kaggle blog is concerned,some part of the talk dealt with bag of words(and other basic vsm techniques) ,embeddings were also introduced as an alternative but the overall aim was to focus on the entire process of starting with raw text and apply ml models to it

**Comment by manasRK** [comment]
You could try using LIME/ELI5 or RF feature importance to showcase cool visuals on how features are interacting and how it helps explain the model.

**Comment by ishaan007** [comment]
Sure, generating a comparative LIME graph for certain predictions by naive bayes and logistic regression( used in the talk) seems a doable thing.Little bit out of bandwidth currently,but will update the slides with it.Thanks for the suggestion,I have proposed an enhanced better version of this for PyData delhi' 17.
